ts/parts/Utilities.ts

```diff
@@ -1427,6 +1444,23 @@ function defined<T>(obj: T): obj is NonNullable<T> {
     return typeof obj !== 'undefined' && obj !== null;
 }
 
+function attr(
+    elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement)
+): undefined;
+function attr(
+    elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement),
+    prop: (Highcharts.HTMLAttributes|Highcharts.SVGAttributes)
+): undefined;
+function attr(
+    elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement),
+    prop: string,
+    value?: undefined
+): (string|null);
+function attr(
+    elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement),
+    prop: string,
+    value: (number|string)
+): undefined;
```

Comment:
You could combine it with the previous declaration to a conditional return type:
```ts
function attr<T extends (number|string|undefined)>(
     elem: (Highcharts.HTMLDOMElement|Highcharts.SVGDOMElement),
     prop: string,
     value?: T
 ): T extends undefined ? (number|string) : undefined;
```

Reply:
I realized that you can not overload this so good, so three declarations are necessary as it seams.